angular/projects/spark-angular/src/lib/components/sprk-tooltip/sprk-tooltip.component.ts

```diff
@@ -0,0 +1,239 @@
+import {
+  Component,
+  AfterViewInit,
+  HostListener,
+  Input,
+  Output,
+  ViewChild,
+  EventEmitter
+} from '@angular/core';
+import { uniqueId } from 'lodash';
+
+@Component({
+  selector: 'sprk-tooltip',
+  template: `
+  <span class="sprk-c-Tooltip__container" #containerElement>
```

Comment:
We want the [attr.data-id]="idString" on the container here instead of the button. I dont think we have this documented anywhere actually, but we place them on the main container because it makes selecting anything inside easier.